tests/Entity/BlockInteractorTest.php

```diff
@@ -13,114 +13,85 @@
 
 namespace Sonata\PageBundle\Tests\Entity;
 
-use Doctrine\Persistence\ManagerRegistry;
-use PHPUnit\Framework\TestCase;
+use Doctrine\ORM\EntityManagerInterface;
 use Sonata\PageBundle\Entity\BlockInteractor;
-use Sonata\PageBundle\Model\Block;
-use Sonata\PageBundle\Model\BlockManagerInterface;
-use Sonata\PageBundle\Model\Page;
 use Sonata\PageBundle\Model\PageInterface;
+use Sonata\PageBundle\Tests\App\AppKernel;
+use Sonata\PageBundle\Tests\App\Entity\SonataPageBlock;
+use Sonata\PageBundle\Tests\App\Entity\SonataPagePage;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 
-final class BlockInteractorTest extends TestCase
+final class BlockInteractorTest extends KernelTestCase
 {
+    /**
+     * @var EntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * @var BlockInteractor
+     */
+    protected $blockInteractor;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $kernel = self::bootKernel();
+
+        $managerRegistry = $kernel->getContainer()->get('doctrine');
+        $blockManager = $kernel->getContainer()->get('sonata.page.manager.block');
+
+        $this->entityManager = $managerRegistry->getManager();
+        $this->blockInteractor = new BlockInteractor($managerRegistry, $blockManager);
+    }
+
     /**
      * @testdox It is returning a block list.
      */
     public function testLoadPageBlocks(): void
     {
-        //Mock
-        $managerRegistryMock = $this->createMock(ManagerRegistry::class);
```

Comment:
Why don't you pass just a mock instead of the service? It should be just a unit test 👀 .